### PR TITLE
Implements local browser chat service

### PR DESCRIPTION
**Pull request by jp848**
**Patch description**
Using the websocket implementation, adds the local browser as a chat service.

**Testing steps**
```
parlai/chat_service/services/browser_chat/run.py --config-path parlai/chat_service/tasks/chatbot/config.yml --port 30003

python client.py --port 30003
```

**Logs**
```
(base) Juui-MacBook-Pro:parlai juuhnpark$ python parlai/chat_service/services/browser_chat/run.py --config-path parlai/chat_service/tasks/chatbot/config.yml --port 30003
[ optional arguments: ] 
[  datapath: /Users/juuhnpark/ParlAI/data ]
[ Chat Services: ] 
[  config_path: parlai/chat_service/tasks/chatbot/config.yml ]
[  is_debug: False ]
[  password: None ]
[ Browser Chat: ] 
[  port: 30003 ]
[ Current ParlAI commit: 1322cdf2bc194ccde851a6f3e209261dff652945 ]
[ warning: overriding opt['model'] to legacy:seq2seq:0 (previously: seq2seq )]
[ Loading existing model params from /Users/juuhnpark/ParlAI/data/models/convai2/seq2seq/convai2_self_seq2seq_model ]
Dictionary: loading dictionary from /Users/juuhnpark/ParlAI/data/models/convai2/seq2seq/convai2_self_seq2seq_model.dict
[ num words =  18803 ]
/anaconda3/lib/python3.7/site-packages/torch/nn/_reduction.py:46: UserWarning: size_average and reduce args will be deprecated, please use reduction='sum' instead.
  warnings.warn(warning.format(ret))
[I 191208 06:50:45 web:2162] 101 GET /websocket (::1) 1.06ms
[I 191208 06:50:45 sockets:35] Opened new socket from ip: ::1
[I 191208 06:50:45 sockets:36] Current subscribers: {'b7558fc7-956e-427e-b348-74ae75966c74': <parlai.chat_service.services.websocket.sockets.MessageSocketHandler object at 0x1a399c8860>}
[I 191208 06:50:59 sockets:55] websocket message from client: {"text": "Hi!"}
[I 191208 06:50:59 agents:40] Sending new message: {'id': 'Overworld', 'text': 'Welcome to the overworld for the ParlAI messenger chatbot demo. Please type "begin" to start.', 'quick_replies': ['begin']}
[I 191208 06:51:02 sockets:55] websocket message from client: {"text": "begin"}
[I 191208 06:51:02 agents:55] Received new message: {'text': 'begin', 'payload': None, 'sender': {'id': 'b7558fc7-956e-427e-b348-74ae75966c74'}, 'recipient': {'id': 0}}

... 

[  port: 10001 ]
[ Current ParlAI commit: 1322cdf2bc194ccde851a6f3e209261dff652945 ]
Connecting to port:  10001
Please connect to the link: http://localhost:8080/
127.0.0.1 - - [08/Dec/2019 06:38:59] "POST /interact HTTP/1.1" 200 -
on_message called
127.0.0.1 - - [08/Dec/2019 06:39:01] "POST /interact HTTP/1.1" 200 -
on_message called
127.0.0.1 - - [08/Dec/2019 06:39:06] "POST /interact HTTP/1.1" 200 -
on_message called

```

**Other information**

Screen Shot 2019-12-08 at 6 51 12 AM
Screenshot from Safari

Screen Shot 2019-12-08 at 6 52 00 AM
Screenshot from Chrome

### CODE REVIEW

**Comment by young-k** [comment]
I have the webhook integration set up properly for the `chatbot` task, but for the other two tasks, it starts up successfully, but I'm having difficulty re-verifying with a different webhook URL. Will update once I'm able to do so, but I can reproduce the same results on `master`, so it's not related to this branch!

`chatbot` task:

Screen Shot 2019-12-08 at 11 02 36 AM

`qada` task:
Screen Shot 2019-12-08 at 11 14 38 AM

`overworld` task:

Screen Shot 2019-12-08 at 11 16 30 AM

**Comment by young-k** [comment]
In the heroku logs, I can see that the webhook is correctly verified, but I'm still not sure why I can't send messages through Messenger:

image

**Comment by young-k** [comment]
Left the server running for 10 minutes and it started working, maybe it was just a long startup time?

proof that functionality is not lost:

Screen Shot 2019-12-08 at 11 27 55 AM

File: parlai/chat_service/services/browser_chat/browser_manager.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+from parlai.chat_service.services.websocket.websocket_manager import WebsocketManager
+
+
+class BrowserManager(WebsocketManager):
+    """
+    Chat Service manager that runs chat service tasks using terminal to send and receive
```
Comment (klshuster) [inline parlai/chat_service/services/browser_chat/browser_manager.py]: nit: update docstring

**Pull request merged**